src/components/Search.tsx

```diff
@@ -1,68 +1,79 @@
-import React from 'react';
-import type {GestureResponderEvent, StyleProp, ViewStyle} from 'react-native';
-import {View} from 'react-native';
-import useLocalize from '@hooks/useLocalize';
-import useTheme from '@hooks/useTheme';
-import useThemeStyles from '@hooks/useThemeStyles';
-import variables from '@styles/variables';
-import CONST from '@src/CONST';
-import Icon from './Icon';
-import * as Expensicons from './Icon/Expensicons';
-import {PressableWithFeedback} from './Pressable';
-import Text from './Text';
-import Tooltip from './Tooltip';
+import React, {useEffect} from 'react';
+import {useOnyx} from 'react-native-onyx';
+import useNetwork from '@hooks/useNetwork';
+import * as SearchActions from '@libs/actions/Search';
+import * as SearchUtils from '@libs/SearchUtils';
+import EmptySearchView from '@pages/Search/EmptySearchView';
+import ONYXKEYS from '@src/ONYXKEYS';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import isLoadingOnyxValue from '@src/types/utils/isLoadingOnyxValue';
+import TableListItemSkeleton from './Skeletons/TableListItemSkeleton';
+
+
+ // For testing purposes run this code in browser console to insert fake data:
+ // query is the param from URL, by default it will be "all"
+ // Onyx.set(`search_${query}`, {
+ //             search: {
+ //                 offset: 0,
+ //                 type: 'transaction',
+ //                 hasMoreResults: false,
+ //             },
+ //             data: {
+ //                 transactions_1234: {
+ //                     receipt: {source: 'http...'},
+ //                     hasEReceipt: false,
+ //                     created: '2024-04-11 00:00:00',
+ //                     amount: 12500,
+ //                     type: 'cash',
+ //                     reportID: '1',
+ //                     transactionThreadReportID: '2',
+ //                     transactionID: '1234',
+ //                 },
+ //                 transactions_5555: {
+ //                     receipt: {source: 'http...'},
+ //                     hasEReceipt: false,
+ //                     created: '2024-04-11 00:00:00',
+ //                     amount: 12500,
+ //                     type: 'cash', // not present in live data (data outside of snapshot_)
+ //                     reportID: '1',
+ //                     transactionThreadReportID: '2',
+ //                     transactionID: '5555',
+ //                 },
+ //             },
+ //         })
```

Comment:
I understand this looks bad, but Im keeping it here for testing purposes to help reviewers. Unfortunately there is no api for this feature done yet.

Once this gets approve I can push a commit to remove this comment.